.github/workflows/unittests.yaml

```diff
@@ -28,21 +31,13 @@ jobs:
           fetch-depth: 0
           lfs: true
       - name: Checkout LFS objects
-        run: git lfs checkout        
-      - name: Setup Scala
-        uses: olafurpg/setup-scala@v12
-        with:
-          java-version: adopt-openj9@1.8.0-292
-      - name: Cache .ivy2
-        uses: actions/cache@v4
-        with:
-          path: ~/.ivy2
-          key: ${{ runner.os }}-ivy2-${{ hashFiles('**/build.sbt') }}
-      - name: Cache .sbt
-        uses: actions/cache@v4
+        run: git lfs checkout
+      - uses: actions/setup-java@v4
         with:
-          path: ~/.sbt
-          key: ${{ runner.os }}-sbt-${{ hashFiles('**/build.properties') }}
+          distribution: 'temurin'
+          java-version: ${{ matrix.java-version }}
+          cache: sbt
```

Comment:
I removed all the manual caching since the `setup-java` action has functionality for sbt-related caching builtin.